src/common/render_tree.cpp

```diff
@@ -118,13 +118,14 @@ static unique_ptr<RenderTreeNode> CreateNode(const PipelineRenderNode &op) {
 static unique_ptr<RenderTreeNode> CreateNode(const ProfilingNode &op) {
 	InsertionOrderPreservingMap<string> extra_info;
 	if (op.GetProfilingInfo().Enabled(MetricsType::EXTRA_INFO)) {
-		extra_info = op.GetProfilingInfo().metrics.extra_info;
+		extra_info = op.GetProfilingInfo().extra_info;
 	}
 
 	auto node_name = op.GetName();
 	auto result = make_uniq<RenderTreeNode>(node_name, extra_info);
-	result->extra_text["Cardinality"] = to_string(op.GetProfilingInfo().metrics.operator_cardinality);
-	string timing = StringUtil::Format("%.2f", op.GetProfilingInfo().metrics.operator_timing);
+	result->extra_text["Cardinality"] = op.GetProfilingInfo().GetMetricAsString(MetricsType::OPERATOR_CARDINALITY);
```

Comment:
While we're updating this method, is it intended that CPU time is left out here?
If not, can we add it in this PR ?

Reply:
CPU time is only calculated at the very end in `QueryProfiler::EndQuery()`

Reply:
Isn't that also where `OPERATOR_TIMING` is calculated?
```c++
			if (query_info.GetProfilingInfo().Enabled(MetricsType::OPERATOR_TIMING)) {
				query_info.GetProfilingInfo().metrics.operator_timing = main_query.Elapsed();
			}
```

Reply:
Only the top level node, `QUERY_INFO`, gets calculated there. All the children get calculated elsewhere. 
We can expand this function to include the other metrics, but I don't think this function currently takes into account whether the metrics are enabled (in profiling settings).

Reply:
I'm going to add this to an issue, for a later fix.